File: scrape-parsing.py
#!/usr/local/bin/python

import json
import glob

in_glob_bills = "_data/mt/bill_*.json"
in_glob_votes = "_data/mt/vote_event_*.json"

out_path_1 = 'data-processed/scrape-2017.json'
out_path_app_votes = 'app/src/data/scrape-2017-votes.json'
out_path_app_bills = 'app/src/data/scrape-2017-bills.json'

# Utility functions
def get_js(path):
    with open(path) as f:
        data = json.load(f)
    return data

def write_js(path, data):
    with open(path, 'w') as f:
      json.dump(data, f, ensure_ascii=False, indent=4)
    
def main():
    bill_files = glob.glob(in_glob_bills)
    vote_files = glob.glob(in_glob_votes)

    bills = [get_js(file) for file in bill_files]
    votes = [get_js(file) for file in vote_files]
    
    write_js(out_path_1, {'bills': bills, 'votes': votes})
    write_js(out_path_app_bills, bills)
    write_js(out_path_app_votes, votes)
    
if __name__ == "__main__":
    main()

File: scrape-parsing.js

/* Run with
    node scrape-parsing.js
*/

const fs = require('fs')
const glob = require('glob')
const moment = require('moment')

const { runTests, processing } = require('./processing/data-processing.js')

const GLOB_BILLS = "_data/mt/bill_*.json"
const GLOB_VOTES = "_data/mt/vote_event_*.json"
// const GLOB_BILLS = "_data/mt-2017/bill_*.json"
// const GLOB_VOTES = "_data/mt-2017/vote_event_*.json"

// const OUT_PATH_APP_VOTES = 'app/src/data/scrape-2019-votes.json'
// const OUT_PATH_APP_BILLS = 'app/src/data/scrape-2019-bills.json'
const OUT_PATH_LAWMAKERS = './analysis/lawmakers-2019.json'
const OUT_PATH_APP = 'app/src/data/mtleg-2019.json'

// Utility functions
const getJson = (path) => JSON.parse(fs.readFileSync(path))

const writeJson = (path, data) => {
    fs.writeFile (path, JSON.stringify(data), function(err) {
        if (err) throw err;
        console.log('Written to', path);
        }
    );
}

function main(){
    console.log('Pulling in', GLOB_BILLS, GLOB_VOTES)
    const bill_files = glob.sync(GLOB_BILLS)
    const vote_files = glob.sync(GLOB_VOTES)

    const bills = bill_files.map(getJson)
    const votes = vote_files.map(getJson)
    const lawmakers = getJson('./app/src/data/leg-roster-2019.json')
    // const lawmakers = getJson('./processing/leg-roster-2017.json')

    // Processing
    // console.log(configs)
 
    processing(bills, votes, lawmakers)
    runTests(bills, votes, lawmakers)

    writeJson(OUT_PATH_APP, {
        updateDate: moment().format('YYYY-MM-DD'),
        votes: votes,
        bills: bills,
        lawmakers: lawmakers,
    })
    writeJson(OUT_PATH_LAWMAKERS, {
        lawmakers: lawmakers,
    })
    // writeJson(OUT_PATH_APP_VOTES, votes)
    // writeJson(OUT_PATH_APP_BILLS, bills)
}
  
main()